libs/go-kibana-rest/kbapi/api._.go

```diff
@@ -128,6 +129,12 @@ func New(c *resty.Client) *API {
 				Delete: newKibanaSyntheticsPrivateLocationDeleteFunc(c),
 				Get:    newKibanaSyntheticsPrivateLocationGetFunc(c),
 			},
+			Parameter: &KibanaSyntheticsParameterAPI{
+				Add:    newKibanaSyntheticsParameterAddFunc(c),
+				Delete: newKibanaSyntheticsParameterDeleteFunc(c),
+				Get:    newKibanaSyntheticsParameterGetFunc(c),
+				Update: newKibanaSyntheticsParameterUpdateFunc(c),
+			},
```

Comment:
We're trying to move everything to a client generated from the Kibana openapi spec. It's not as smooth as I'd like at the moment, but I've done a bit of the work to get started in [this](https://github.com/elastic/terraform-provider-elasticstack/commit/32df9f794d0ba92bee5c3a2be183fcb3d227b7aa) commit. Would you be able to migrate this PR over to that client instead of updating this one? 

Given the delete endpoint you're using here is undocumented we may need to leave that code in here, but at least for get/post/put we should be able to use the generated client.

Reply:
Sorry for taking ages here. The failures in this PR are all fixed by https://github.com/elastic/terraform-provider-elasticstack/pull/1189. We should just be able to merge main when it's fixed.